src/ds/json/adv-jsonpath.md

```diff
@@ -0,0 +1,191 @@
+[JSONPath](https://goessner.net/articles/JsonPath/) expressions help you access specific elements within a JSON document, which is similar to how XPATH works for XML documents.
+JSONPath support was added to Redis Stack in version 2.0.
+Before that, [a legacy form of pathing](https://redis.io/docs/data-types/json/path/#legacy-path-syntax) was supported.
+Only JSONPath will be discussed in this tutorial.
+
+You've already seen several examples of JSONPath in previous parts of this tutorial. The next sections will describe JSONPath in more complete detail.
+Some simple JSON documents will be used to demonstrate JSONPath's features.
+
+```redis Load documents
+JSON.SET lit1 $ 5
+JSON.SET lit2 $ '"abc"'
+JSON.SET lit3 $ true
+JSON.SET lit4 $ null
+JSON.SET obj1 $ '{"a":1, "b":2}'
+JSON.SET obj2 $ '{"a":[1,2,3,"a","b","c",false,true,["a",1],{"a":1},{"b":null}], "b":5}'
+JSON.SET arr1 $ [1,2,3]
+JSON.SET arrmap $ '[{"a":1}, {"b":2}]'
+```
+
+The returned value of a JSONPath expression is a JSON string with a top-level array of JSON serialized strings.
+
+**Note**:
+>A JSONPath query can resolve to several locations in a JSON document. When more than one location is identified, the JSON commands apply the operation to every possible location.
+
+## JSONPath syntax
+
+| Syntax&nbsp;element | Description |
+|----------------|-------------|
+| `$` | The root (outermost JSON element), starts the path. |
+| `.` or `[]` | Selects a child element. |
+| `..` | Recursively descends through the JSON document. |
+| `*` | Wildcard, returns all elements. |
```

Comment:
If the current node is an array - `*` resolves to all the array’s elements
If the current node is an object - `*` resolves to all the object's members